vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/SimpleAuthorizationService.kt

```diff
@@ -145,16 +145,13 @@ class SimpleAuthorizationService(
         } ?: throw OAuth2Exception(Errors.INVALID_REQUEST)
             .also { Napier.w("token: could not load user info for $params}") }
 
-        if (params.authorizationDetails != null) {
-            // TODO verify params.authorizationDetails.claims and so on
-            params.authorizationDetails.credentialIdentifiers?.forEach { credentialIdentifier ->
-                // TODO work out mapping of credential identifiers in authorization details to schemes
-                if (!credentialSchemes.map { it.vcType }.contains(credentialIdentifier)) {
-                    throw OAuth2Exception(Errors.INVALID_GRANT)
-                        .also { Napier.w("token: client requested invalid credential identifier: $credentialIdentifier") }
-                }
-            }
-        }
+        // TODO work out mapping of credential identifiers in authorization details to schemes
+        val filteredAuthorizationDetails = params.authorizationDetails?.filter {
+            credentialSchemes.map { it.vcType }.contains(it.credentialConfigurationId) ||
+            credentialSchemes.map { it.sdJwtType }.contains(it.credentialConfigurationId)
```

Comment:
possible typo? Missing || at end of line, effectively making the first two lines useless